doc/Language/regexes.pod6

```diff
@@ -512,10 +512,43 @@ string of non-whitespace characters.
 
 =head1 X<Longest Alternation: C<|>|regex,|>
 
-In regexes branches separated by C<|>, the longest match wins, independent of
-the lexical ordering in the regexes.
+For short, in regexes branches separated by C<|>, the longest token match wins,
+independent of the textual ordering in the regexes. However, what C<|> really
+does is further more.
 
-    say ('abc' ~~ / a | .b /).Str;    # OUTPUT: «ab␤»
+C<|> does not decide which breanch to win after finish the whole match,
+but follows L<longest token match, for short, LTM|
+https://design.perl6.org/S05.html#Longest-token_matching> strategy.
+
+Briefly, what C<|> does is this:
+
+=item1 First, select the branch who has the longest token prefix.
+
+=item1 If it's a tie, select the match with the highest specificity.
+
+=item1 If it's still a tie, use additional tie-breakers.
+
+    say "abc" ~~ /ab | a.* /;                 # Output: ⌜abc⌟
+    say "abc" ~~ /ab | a {} .* /;             # Output: ⌜ab⌟
+    say "abc" ~~ /a. | ab { print "win" } /;  # Output: win｢ab｣
+    say "abc" ~~ /a\w| a. { print "lose" } /; # Output: ⌜ab⌟
+    say "if else" ~~ / if | if <.ws> else /;  # Output: ｢if｣
+    say "if else" ~~ / if | if \s+   else /;  # Output: ｢if else｣
+
+As the first pair of example above C<a.*> is a token prefix, while C<a {} .*>
+terminates at C<{}>, then its token prefix is C<a>.
+
+The second pair, when two alternatives match at the same length, the tie is
```

Comment:
Maybe this would be easier to read if you moved the examples closer to the explanations, by splitting up the examples.

Digesting 6 examples at once is quite a bit, and then later on you have to start counting to see which examples the explanation talks about

Reply:
@moritz Following your advice, I separate the examples.

Actually I rearrange the text, does it read smoothly now?